### PR TITLE
Migrate tekton results e2e test to kind

### PR DESCRIPTION
**Pull request by enarha**
# Changes

Migrate Tekton Results e2e test to kind.
Roughly based on #1252 and https://github.com/tektoncd/plumbing/blob/main/docs/kind-e2e.md

/kind misc

Signed-off-by: Emil Natan <ena@redhat.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Comment by adambkaplan** [comment]
@tektoncd/plumbing-maintainers this is ready for review.

**Review by vdemeester — APPROVED** [review]

File: prow/config.yaml
Comment (adambkaplan) [inline prow/config.yaml:1604]: Documenting the chain that leads to the e2e tests running:

1. presubmit-tests script is defined in [tektoncd/results/test](https://github.com/tektoncd/results/blob/main/test/presubmit-tests.sh)
2. This calls the [plumbing presubmit-tests script](https://github.com/tektoncd/plumbing/blob/main/scripts/presubmit-tests.sh#L244)
3. The plumbing script looks for any script in tektoncd/results that matches `e2e-*test.sh` - [code](https://github.com/tektoncd/plumbing/blob/main/scripts/presubmit-tests.sh#L275-L281)
4. The e2e test script in results calls a [setup](https://github.com/tektoncd/results/blob/main/test/e2e/e2e.sh#L38) [script](https://github.com/tektoncd/results/blob/main/test/e2e/00-setup.sh#L22) that creates the kind cluster. This lets us define the version of kind we use in-tree.

**Review by adambkaplan — APPROVED** [review]
/lgtm

Noting that the e2e script in results creates the kind cluster. There is no need to use the test runner's command to configure kind.

**Comment by adambkaplan** [comment]
@tektoncd/plumbing-collaborators @tektoncd/plumbing-maintainers I believe we need another reviewer to lgtm this change.

**Pull request merged**
